source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -57,72 +59,108 @@ struct SuccessResponse {
-  std::vector<Matchers::LowerCaseStringMatcherPtr> matchers;
+envoy::type::matcher::v3::StringMatcher
+ignoreCaseStringMatcher(const envoy::type::matcher::v3::StringMatcher& matcher) {
+  envoy::type::matcher::v3::StringMatcher ignore_case;
+  ignore_case.set_ignore_case(true);
+  switch (matcher.match_pattern_case()) {
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kSafeRegex:
+    return matcher;
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kHiddenEnvoyDeprecatedRegex:
+    ignore_case.set_hidden_envoy_deprecated_regex(matcher.hidden_envoy_deprecated_regex());
+    break;
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kExact:
+    ignore_case.set_exact(matcher.exact());
+    break;
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kPrefix:
+    ignore_case.set_prefix(matcher.prefix());
+    break;
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kSuffix:
+    ignore_case.set_suffix(matcher.suffix());
+    break;
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+  return ignore_case;
+}
+
+std::vector<Matchers::StringMatcherPtr>
+createStringMatchers(const envoy::type::matcher::v3::ListStringMatcher& list,
+                     const bool disable_lowercase_string_matcher) {
+  std::vector<Matchers::StringMatcherPtr> matchers;
   for (const auto& matcher : list.patterns()) {
-    matchers.push_back(std::make_unique<Matchers::LowerCaseStringMatcher>(matcher));
+    matchers.push_back(std::make_unique<Matchers::StringMatcherImpl>(
+        disable_lowercase_string_matcher ? matcher : ignoreCaseStringMatcher(matcher)));
   }
   return matchers;
 }
 
 } // namespace
 
 // Matchers
-HeaderKeyMatcher::HeaderKeyMatcher(std::vector<Matchers::LowerCaseStringMatcherPtr>&& list)
+HeaderKeyMatcher::HeaderKeyMatcher(std::vector<Matchers::StringMatcherPtr>&& list)
     : matchers_(std::move(list)) {}
 
 bool HeaderKeyMatcher::matches(absl::string_view key) const {
   return std::any_of(matchers_.begin(), matchers_.end(),
                      [&key](auto& matcher) { return matcher->match(key); });
 }
 
-NotHeaderKeyMatcher::NotHeaderKeyMatcher(std::vector<Matchers::LowerCaseStringMatcherPtr>&& list)
+NotHeaderKeyMatcher::NotHeaderKeyMatcher(std::vector<Matchers::StringMatcherPtr>&& list)
     : matcher_(std::move(list)) {}
 
 bool NotHeaderKeyMatcher::matches(absl::string_view key) const { return !matcher_.matches(key); }
 
 // Config
 ClientConfig::ClientConfig(const envoy::extensions::filters::http::ext_authz::v3::ExtAuthz& config,
                            uint32_t timeout, absl::string_view path_prefix)
-    : request_header_matchers_(
-          toRequestMatchers(config.http_service().authorization_request().allowed_headers())),
-      client_header_matchers_(toClientMatchers(
-          config.http_service().authorization_response().allowed_client_headers())),
+    : disable_lowercase_string_matcher_(Runtime::runtimeFeatureEnabled(
```

Comment:
nit: can you flip the name of the member variable to match the default true state (and the runtime key)?

Reply:
Done.